mypy/partially_defined.py

```diff
@@ -314,11 +311,20 @@ class PossiblyUndefinedVariableVisitor(ExtendedTraverserVisitor):
     """
 
     def __init__(
-        self, msg: MessageBuilder, type_map: dict[Expression, Type], options: Options
+        self,
+        msg: MessageBuilder,
+        type_map: dict[Expression, Type],
+        options: Options,
+        names: SymbolTable,
     ) -> None:
         self.msg = msg
         self.type_map = type_map
         self.options = options
+        self.builtins: set[str] = set()
+        builtins_mod = names.get("__builtins__", None)
+        if builtins_mod:
+            assert isinstance(builtins_mod.node, MypyFile)
+            self.builtins = set(builtins_mod.node.names.keys())
```

Comment:
As a minor optimization, we could store `builtins_mod.node.names` in `self.builtins` (i.e. a `SymbolTable` instance). This would avoid having to create a set with all the items in builtins.